### PR TITLE
ci: migrate test builds to github actions

### PR DESCRIPTION
**Pull request by r-rojo**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -46,51 +46,6 @@ matrix:
...
-    - TRAVIS_BUILD_TYPE=Release
-    addons:
-      apt:
-        packages:
-        - libjson-c-dev
-        - libhwloc-dev
-        - libtbb-dev
-        - uuid-dev
-        - doxygen
-  - script: scripts/test-build.sh
```
Comment (asgardkm) [inline .travis.yml:79]: @r-rojo will this change depreciate the `scripts/test-build.sh` script?

File: .travis.yml [hunk 2]
```diff
@@ -46,51 +46,6 @@ matrix:
       distributions: sdist
       on:
         tags: true
-  - script: scripts/test-build.sh
-    install:
-    - pip install --user pybind11 nose2
```
Comment (asgardkm) [inline .travis.yml:51]: @r-rojo I don't see pip dependencies ported over to the GitHub CI. Are they still needed?

File: .github/workflows/ccpp-build.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: C/C++ CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
```
Comment (asgardkm) [inline .github/workflows/ccpp-build.yml:9]: @r-rojo nice :)

File: .github/workflows/ccpp-build.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    - uses: actions/checkout@v2
+    - name: update
+      run: sudo apt-get update -y
+    - name: Get Packages
+      uses: mstksg/get-package@v1
+      with:
+        apt-get: uuid-dev libjson-c-dev libhwloc-dev lcov libtbb-dev
+    - name: configure ${{ matrix.build-type }}
+      run: mkdir ${{ github.workspace }}/.build && cd ${{ github.workspace }}/.build && cmake .. -DCMAKE_BUILD_TYPE=${{ matrix.build-type }}
+    - name: make ${{ matrix.build-type }}
```
Comment (asgardkm) [inline .github/workflows/ccpp-build.yml:25]: @r-rojo do you see value in generating `deb` packages in same flow?

I only see `ubuntu` supported by GitHub's default virtual environments, so unfortunately can't natively extend functionality to `centos`/`fedora` VMs.

File: .travis.yml (hunk 2 above)
Comment (asgardkm) [reply .travis.yml:51]: Probably not, seeing latest CI run: https://github.com/OPAE/opae-sdk/runs/930116673?check_suite_focus=true

File: .travis.yml (hunk 1 above)
Comment (asgardkm) [reply .travis.yml:79]: If so, could/should it be removed?

Comment (r-rojo) [reply .travis.yml:79]: Yes, and yes

File: .github/workflows/ccpp-build.yml (hunk 4 above)
Comment (r-rojo) [reply .github/workflows/ccpp-build.yml:25]: I don't recall travis supporting centos/fedora either. These are just sanity build checks. We can/should revisit packaging later though.

Comment (asgardkm) [reply .github/workflows/ccpp-build.yml:49]: ack

File: .github/workflows/ccpp-tests.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        apt-get: uuid-dev libjson-c-dev libhwloc-dev lcov libtbb-dev
+    - name: set hugepages
+      run: sudo sysctl -w vm.nr_hugepages=8
+    - name: run ${{ matrix.test-script }}
+      run: ${{ github.workspace }}/scripts/${{ matrix.test-script }}
+    - name: Coveralls
+      if: ${{ matrix.test-script == 'cover.sh' }}
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.github_token }}
```
Comment (asgardkm) [inline .github/workflows/ccpp-tests.yml:55]: @r-rojo I don't see the coveralls reports appearing in the CI.

Perhaps `secrets.github_token` is case sensitive?
https://github.com/marketplace/actions/coveralls-github-action

**Pull request merged**